src/util/pg.py

```diff
@@ -1048,21 +1048,29 @@ def rename_table(cr, old_table, new_table, remove_constraints=True):
         """,
         [new_table],
     )
-    (old_pkey,) = cr.fetchone()
-    new_pkey = new_table + "_pkey"
-    cr.execute(sql.SQL("ALTER INDEX {} RENAME TO {}").format(sql.Identifier(old_pkey), sql.Identifier(new_pkey)))
+    try:
+        (old_pkey,) = cr.fetchone()
+    except TypeError:
+        pass  # No primary key found
+    else:
+        new_pkey = new_table + "_pkey"
+        cr.execute(
+            sql.SQL("ALTER INDEX {} RENAME TO {}").format(
+                sql.Identifier(old_pkey), sql.Identifier(new_pkey)
+            )
+        )
 
-    # rename indexes (except the pkey's one and those not containing $old_table)
-    cr.execute(
-        """
-        SELECT indexname
-          FROM pg_indexes
-         WHERE tablename = %s
-           AND indexname != %s
-           AND indexname LIKE %s
-        """,
-        [new_table, new_pkey, "%" + old_table.replace("_", r"\_") + r"\_%"],
-    )
+        # rename indexes (except the pkey's one and those not containing $old_table)
+        cr.execute(
+            """
+            SELECT indexname
+            FROM pg_indexes
+            WHERE tablename = %s
+            AND indexname != %s
+            AND indexname LIKE %s
+            """,
+            [new_table, new_pkey, "%" + old_table.replace("_", r"\_") + r"\_%"],
+        )
```

Comment:
The indexes can still be renamed, even if no PKey.
```suggestion
    else:
        new_pkey = ""  # no PK renamed
    # rename indexes (except the pkey's one and those not containing $old_table)
    cr.execute(
        """
        SELECT indexname
        FROM pg_indexes
        WHERE tablename = %s
        AND indexname != %s
        AND indexname LIKE %s
        """,
        [new_table, new_pkey, "%" + old_table.replace("_", r"\_") + r"\_%"],
    )
```